test/box-luatest/pagination_netbox_test.lua

```diff
@@ -0,0 +1,373 @@
+        t.assert_equals(tuples1, tuples_offset)
+        t.assert_equals(tuples2, tuples_offset)
+    end
+    data_size = s:select(nil,
+            {limit=2, fetch_pos=true, after=pos, buffer=ibuf})
+    data = mp.decode(ibuf.rpos, data_size)
+    tuples1 = data[48]
+    pos = data[0x35]
+    t.assert_equals(tuples1, {})
+    t.assert_equals(pos, nil)
+    ibuf:reset()
+end
+
+net_g.test_net_box_skip_header = function(cg)
+    cg.server:exec(function()
+        local s = box.space.s
+        s:create_index("pk", {type = "tree"})
+        s:create_index("sk", {parts = {2}, type = "tree", unique=false})
+    end)
+    local net_box = require('net.box')
+    local conn = net_box.connect(cg.server.net_box_uri)
+    local s = conn.space.s
+
+    local mp = require('msgpack')
+    local buffer = require('buffer')
+    local ibuf = buffer.ibuf()
+
+    -- Fetch position in empty space
+    local data_size, pos = s:select(nil,
+            {limit=2, fetch_pos=true, buffer=ibuf, skip_header=true})
+    local data = mp.decode(ibuf.rpos, data_size)
+    t.assert_equals(data, {})
+    t.assert_equals(pos, nil)
+    ibuf:reset()
+    cg.server:exec(function()
+        for i = 1, 11 do
+            box.space.s:replace{i, 1}
+        end
+    end)
+    local tuples1
+    local tuples2
+    local tuples_offset
+    local pos = ""
+    local last_tuple = box.NULL
+
+    -- Test fullscan pagination
+    for i = 0, 5 do
+        data_size, pos = s:select(nil,
+            {limit=2, fetch_pos=true, after=pos, buffer=ibuf, skip_header=true})
+        tuples1 = mp.decode(ibuf.rpos, data_size)
+        t.assert_not_equals(pos, nil)
+        ibuf:reset()
+        data_size = s:select(nil,
+            {limit=2, after=last_tuple, buffer=ibuf, skip_header=true})
+        tuples2 = mp.decode(ibuf.rpos, data_size)
+        ibuf:reset()
+        last_tuple = tuples2[#tuples2]
+        tuples_offset = s:select(nil, {limit=2, offset=i*2})
+        t.assert_equals(tuples1, tuples_offset)
+        t.assert_equals(tuples2, tuples_offset)
+    end
+    data_size, pos = s:select(nil,
+            {limit=2, fetch_pos=true, after=pos, buffer=ibuf, skip_header=true})
+    tuples1 = mp.decode(ibuf.rpos, data_size)
+    t.assert_equals(tuples1, {})
+    t.assert_equals(pos, nil)
+    ibuf:reset()
+end
+
+net_g.test_net_box_invalid_position = function(cg)
+    cg.server:exec(function()
+        local s = box.space.s
+        s:create_index('pk', {parts={{field=1, type ='uint'}}, type ='tree'})
+        s:create_index('sk', {
+                parts = {{field = 2, type = 'uint', is_nullable=true}},
+                type = 'tree', unique=false})
+    end)
+    local net_box = require('net.box')
+    local conn = net_box.connect(cg.server.net_box_uri)
+    local s = conn.space.s
+    local sk = s.index.sk
+    s:replace{1, 0}
+    s:replace{2, 0}
+    local _, pos = sk:select(nil, {limit=1, fetch_pos=true})
+    t.assert_error_msg_contains("Invalid key part count",
```

Comment:
Nit: Please use `assert_error_msg_equals` or `assert_error_msg_content_equals` in the tests. There's no need to check only part of the error message.

Reply:
DOne